### PR TITLE
Fix address duplication on address completion form.

### PR DESCRIPTION
**Pull request by alvim**
#### What problem is this solving?

The address was being duplicated in address completion form.

#### How should this be manually tested?

[Workspace](https://chk127--checkoutio.myvtex.com/)
Proceed to checkout. Insert a new address. Select delivery option. Complete address information (number, etc). After that, go back to addresses list. There should be only one item of your last registered address.

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [ ] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
_ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: react/shippingStateMachine/useShippingStateMachine.ts [hunk 1]
```diff
@@ -75,8 +74,15 @@ const useShippingStateMachine = ({
...
         })
       },
-      tryToCreateAddress: (_, event) => {
-        return insertAddress(event.address)
+      tryToCreateAddress: async (_, event) => {
+        const res = await insertAddress(event.address)
+        const { success, orderForm } = res;
+
+        if (success && orderForm?.shipping.selectedAddress) {
+          setAddress(orderForm.shipping.selectedAddress)
```
Comment (lucasecdb) [inline react/shippingStateMachine/useShippingStateMachine.ts:82]: do we need to add this call to other methods as well? such as `tryToSelectAddress`, etc. I know I had to add it to `updateSelectedAddress` but I wonder if we need to add it to the other ones

**Review by lucasecdb — APPROVED** [review]
nice!! looks good, left a question

File: react/shippingStateMachine/useShippingStateMachine.ts (hunk 1 above)
Comment (alvim) [reply react/shippingStateMachine/useShippingStateMachine.ts:82]: I'll do some reviewing... I think I was working with a branch created from an old master. When I just tried to create this PR, I had to do a rebase and I've pulled your code from some days ago that do this. And I'm wondering exactly this too.

I'll review the code and rerun this. I'll post any update here soon.

Comment (lucasecdb) [reply react/shippingStateMachine/useShippingStateMachine.ts:82]: nice! thanks

**Review by marcoskwkm — APPROVED** [review]
lgtm

**Pull request closed without merging**